test/tools.cpp

```diff
@@ -36,35 +36,55 @@ namespace unittests
 {
 
 TempFile::TempFile(const char* name)
+ : fd_(-1)
 {
 #ifdef _WIN32
   std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> utfConv;
   wchar_t cbase[MAX_PATH];
-  wchar_t ctmp[MAX_PATH];
   const std::wstring wname = utfConv.from_bytes(name);
   GetTempPathW(MAX_PATH-(wname.size()+2), cbase);
-  // This create a file for us, ensure it is unique.
-  // So we need to delete it and create the directory using the same name.
-  GetTempFileNameW(cbase, wname.c_str(), 0, ctmp);
-  auto tmp_fd = _wopen(ctmp, _O_CREAT | _O_TEMPORARY | _O_SHORT_LIVED | _O_RDWR | _O_TRUNC);
-  path_ = utfConv.to_bytes(ctmp);
+  //This create a empty file, we just have to open it later
+  GetTempFileNameW(cbase, wname.c_str(), 0, wpath_);
+  path_ = utfConv.to_bytes(wpath_);
 #else
   const char* const TMPDIR = std::getenv("TMPDIR");
   const std::string tmpdir(TMPDIR ? TMPDIR : "/tmp");
   path_ = tmpdir + "/" + name + "_XXXXXX";
   auto tmp_fd = mkstemp(&path_[0]);
+  ::close(tmp_fd);
 #endif
-  fd_ = tmp_fd;
 }
 
 TempFile::~TempFile()
 {
-  close(fd_);
-#ifndef _WIN32
+  close();
+#ifdef _WIN32
+  DeleteFileW(wpath_);
+#else
```

Comment:
You don't need this if you `_wopen()` in `TempFile::fd()` using the `_O_TEMPORARY` mode.